src/View/ViewableData.php

```diff
@@ -28,6 +28,7 @@
  * is provided and automatically escaped by ViewableData. Any class that needs to be available to a view (controllers,
  * {@link DataObject}s, page controls) should inherit from this class.
  */
+#[\AllowDynamicProperties]
```

Comment:
What specifically is this resolving that we can't resolve by using non-deprecated functionality?
Adding this is effectively kicking the can down the road for when they remove this attribute.

Reply:
Apologies, forgot to put explainer on this one

It's for __set() which calls `setField()` which calls `$this->$field = $value;`

`getField()` and `hasField()` both also use `$this->$field`

I was worried about the consequences of changing this, though I'll update the PR to use $this->data[$name] instead